editor/plugins/tiles/tile_set_atlas_source_editor.cpp

```diff
@@ -872,9 +872,9 @@ void TileSetAtlasSourceEditor::_update_current_tile_data_editor() {
 	// Find the property to use.
 	String property;
 	if (tools_button_group->get_pressed_button() == tool_select_button && tile_inspector->is_visible() && !tile_inspector->get_selected_path().is_empty()) {
-		Vector<String> components = tile_inspector->get_selected_path().split("/");
-		if (components.size() >= 1) {
-			property = components[0];
+		const String &selected = tile_inspector->get_selected_path();
```

Comment:
I think I would avoid changing it here for consistency. The file uses this pattern everywhere:
```
Vector<String> components = String(p_property).split("/", true, 2);
```
Not that I mind much, but I think consistency kind of helps maintenance and refactoring.

Reply:
I don't really care about consistency if it means creating a temporary array just to fetch 1 element. Also I don't see how this helps.

Reply:
I mean, I do... Code is a lot simpler to follow when it reuse similar patterns in similar cases.

To be clear, I don't mind much the change to be honest. It just feel like this is basically some unneeded optimization (at least in that place I think it brings no significant performance improvement, in the other cases I don't know). But well, it's not a big deal, we can merge as-is it's ok too.

Reply:
I agree that consistency is valuable in such big systems. What are the other cases in this file? Perhaps everything can switch to using get_slice()?

Reply:
Well, technically my changes are self-consistent. `get_slice()` is used everywhere where no more than 2 sliced values are accessed.

Reply:
I searched for where I used this pattern (with the components vector), this is the list:
![Copie d'écran_20240813_160941](https://github.com/user-attachments/assets/451db4e2-6d4b-4a1e-847f-ac704f46f308)

To be fair, I think the change is OK as part of such a big refactoring PR. But I can't promise to think about keeping things that way the next time I have to refactor something in the TileMap/Set area. :sweat_smile:

Reply:
@KoBeWi Do you want to switch these cases over to get_slice? too, then we can have consistency and performance :)

Reply:
They all use the `maxsplit` argument of `split()`, which don't exist in `get_slice()`. The inspector one loops over the strings, so slicing is justified.

Looks like the code here is the only component split this way 🤔